utils/tzktHooks.ts

```diff
@@ -0,0 +1,152 @@
+import { useEffect, useState } from "react";
+import { TZKT_API_URL } from "../context/config";
+
+/**
+ * Returns the balance of the given address
+ * @param address tz/kt Tezos address
+ */
+export const useTzktBalance = (address: string | null) => {
+  const [balance, setBalance] = useState(0);
+  useEffect(() => {
+    if (address == null) return;
+    (async () => {
+      const response = await fetch(
+        `${TZKT_API_URL}/v1/accounts/${address}/balance`
+      );
+
+      if (response.status == 200) {
+        const json: number = await response.json();
+        setBalance(json / 1_000_000); // Divided by the XTZ decimal
+      }
+    })();
+  }, [address]);
+
+  return balance;
+};
+
+/**
+ * Token type returned by:
+ * https://api.tzkt.io/v1/tokens/balances
+ * Only useful fields are typed
+ */
+type tzkt_token = {
+  contract: {
+    address: string;
+  };
+  token: {
+    contract: {
+      address: string;
+    };
+    metadata: {
+      name: string | null;
+      symbol: string | null;
+      decimals: string | null;
+      thumbnailUri: string | null; // this is not documented in the Tzkt API
+    } | null;
+  };
+  balance: string;
+};
+
+export type defi = {
+  contract: string;
+  balance: number;
+  symbol: string;
+  icon: string;
+};
+
+/**
+ * Returns Defi Tokens of the given address
+ * @param address tz/kt Tezos address
+ */
+export const useTzktDefiTokens = (address: string | null) => {
+  const [defi, setDefi] = useState<defi[]>([]);
+
+  useEffect(() => {
+    if (address == null) return;
```

Comment:
`===`

Reply:
fixed in: https://github.com/marigold-dev/tzsafe-ui/pull/167/commits/1491ff68e23888cdbc315a329c0c95e66375bf85